pkg/multimodelconfig/configmap.go

```diff
@@ -0,0 +1,128 @@
+package multimodelconfig
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1beta1"
+	"github.com/kubeflow/kfserving/pkg/constants"
+	"io"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"strings"
+)
+
+var logger = log.Log.WithName("MultiModelConfig")
+
+type ModelConfig map[string]*v1beta1.ModelSpec
+
+type ConfigsDelta struct {
+	updated ModelConfig
+	deleted ModelConfig
+}
+
+func NewConfigsDelta(updatedCfg ModelConfig, deletedCfg ModelConfig) *ConfigsDelta {
+	return &ConfigsDelta{updated: updatedCfg, deleted: deletedCfg}
+}
+
+//multi-model ConfigMap
+//apiVersion: v1
+//kind: ConfigMap
+//metadata:
+//  name: models-config
+//  namespace: <user-model-namespace>
+//data:
+//  example_models.json: |
+//    {
+//      example_model_name1: {
+//        storageUri: s3://example-bucket/path/to/model_name1
+//        framework: sklearn
+//      },
+//      example_model_name2: {
+//        storageUri: s3://example-bucket/path/to/model_name2
+//        framework: sklearn
+//      }
+//    }
+
+func (config *ConfigsDelta) Process(configMap *v1.ConfigMap) (err error) {
+	if err = config.apply(configMap); err != nil {
+		return err
+	}
+	if err = config.delete(configMap); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (config *ConfigsDelta) apply(configMap *v1.ConfigMap) (err error) {
+	if len(config.updated) == 0 {
+		return nil
+	}
+	if configMap.Data == nil {
+		configMap.Data = map[string]string{}
+	}
+	if data, err := decode(configMap.Data[constants.MultiModeConfigFileName]); err == nil {
```

Comment:
```suggestion
	if data, err := decode(configMap.Data[constants.ModelConfigFileName]); err == nil {
```

As Ellis suggested in pr https://github.com/kubeflow/kfserving/pull/1013, let's change MultiModeConfigFileName o ModelConfigFileName